contrib/go.mongodb.org/mongo-driver.v2/mongo/option.go

```diff
@@ -37,3 +39,15 @@ func WithService(name string) OptionFn {
 		cfg.serviceName = name
 	}
 }
+
+// WithMaxQuerySize sets the maximum query size (in bytes) before queries
+// are truncated when attached as a span tag.
+//
+// If max is < 0, the query is never truncated.
+//
+// Defaults to -1.
```

Comment:
Following with my previous comments, I'd rather to make `0` value something more meaningful, and default to a higher max length instead of not truncating. As long as it's configurable, it's ok to do this kind of changes.

The question is "is it possible to have an empty query?". If not, I'd say that `0` should mean the same as a negative value. If it's possible, then `0` should set `""` as tag value. WDYT?

Reply:
If y'all want to avoid truncation here (e.g. due to the obfuscation logic [mentioned above](https://github.com/DataDog/dd-trace-go/pull/3679#issuecomment-3020608781)), then another option is to drop the `query` tag if the length is beyond the configurable limit. Therefore, you always get a well-formed query.